insights/specs/datasources/leapp.py

```diff
@@ -0,0 +1,61 @@
+"""
+Custom datasources for leapp
+"""
+import json
+
+from insights.core.context import HostContext
+from insights.core.exceptions import SkipComponent
+from insights.core.plugins import datasource
+from insights.core.spec_factory import DatasourceProvider, simple_file
+from insights.specs import Specs
+
+
+class LocalSpecs(Specs):
+    leapp_report_raw = simple_file("/var/log/leapp/leapp-report.json")
+    """ Returns the contents of the file ``/var/log/leapp/leapp-report.json`` """
+
+
+@datasource(LocalSpecs.leapp_report_raw, HostContext)
+def leapp_report(broker):
+    """
+    This datasource get useful information from ``leapp-report.json``.
+
+    .. note::
+        Since this file may contain sensitive information, this datasource does
+        filter and only keep required data.
+
+    Returns:
+        str: JSON string after get the required data.
+
+    Raises:
+        SkipComponent: When the path does not exist, nothing is collected,
+                       or any exception occurs.
+    """
+    relative_path = 'insights_commands/leapp_report'
+    valid_keys = {
+        'inhibitor': ['title', 'summary', ('detail', 'remediations')]
+    }
+    try:
+        content = broker[LocalSpecs.leapp_report_raw].content
+        report = json.loads('\n'.join(content))
+        if isinstance(report, dict):
+            results = []
+            for entry in report.get('entries', []):
+                for flag, keys in valid_keys.items():
+                    if flag in entry['flags']:
```

Comment:
You couldn't know and it occurred to me only this morning, leapp tool is deprecating the 'flags' key  in favor of new 'groups' ('flags' + 'tags'), **the schema with 'flags' will still be default for all customers**,  but to make it work in future, could this be changes in a way that if 'flags' key is not present then the parser will check 'groups'  key?

meaning it would work  for this entry
```
{
    .... other data
    'flags': ['inhibitor']
    .... other data
}
``` 
and also this entry  (only difference is that in this second one 'flags' and 'tags' keys are not present, instead 'groups' key is
```
{
    .... other data
    'groups': ['inhibitor']
    .... other data
}
``` 

the mentioned change is

Reply:
Just sharing hack I used in other script this morning 
```py
entry_flags = entry.get("flags", entry.get("groups", []))
```

Reply:
Thanks for the information. Updated.